corehq/apps/app_manager/migrations/0021_migrate_case_search_itemset_ids.py

```diff
@@ -10,7 +10,8 @@ class Migration(migrations.Migration):
     ]
 
     operations = [
-        run_once_off_migration(
-            'migrate_case_search_prompt_itemset_ids', required_commit='116a6fc9556ac929e29a35fd2dfa673984820561'
+        prompt_for_historical_migration(
+            'app_manager', get_migration_name(__file__),
+            required_commit='116a6fc9556ac929e29a35fd2dfa673984820561'
```

Comment:
What do you think of just passing in `__file__` here and have `prompt_for_historical_migration` call `get_migration_name` on it internally?
```suggestion
            'app_manager', __file__, required_commit='116a6fc9556ac929e29a35fd2dfa673984820561'
```

Reply:
Yeah, I'm mixed on this. It bothers me that `__file__` needs to be passed in at all. I thought about examining the stack within `prompt_for_historical_migration` to get the caller's module, but felt that was too hacky. In this case, I think it's probably better to leave it as is? I prefer functions to use the least information necessary, and `prompt_for_historical_migration` does not care about a full file path; all it cares about is the module name. It makes the function easier to call, but violates that rule of least information.

Reply:
My suggestion was optimizing for making it easier to call. That seems more important than the internal purity of `prompt_for_historical_migration`. The usage pattern predicts that there will be many more calls to that function than there will be people critiquing its implementation.

Reply:
It's not just the internal purity, but also whether or not the call itself makes sense. It's not particularly obvious, without looking at the implementation, why `prompt_for_historical_migration` would want the caller's filename, whereas I think its more obvious why the migration name would be required. I don't expect this command to be used that frequently, so I'm fine with making calling it slightly inconvenient. If you think it will be a big enough issue, I'd prefer to just drop the migration name entirely and just tell the user to run `./manage.py migrate --fake {app_name} <migration name}` and assume the user can figure out the migration that failed.

Reply:
> I don't expect this command to be used that frequently, so I'm fine with making calling it slightly inconvenient.

Do you mean that you don't expect developers to be adding new calls to this code frequently, or that you don't expect people to hit this prompt frequently?

> assume the user can figure out the migration that failed

If we can figure out the migration name - which clearly we can - we should do so, instead of adding more burden on the user. This is already not the best experience for the end user.

Reply:
> Do you mean that you don't expect developers to be adding new calls to this code frequently, or that you don't expect people to hit this prompt frequently?

The former. It might just be the case that we have so few calls to the old version because it was relatively new, but removing code that old migrations depend on doesn't seem like it will occur that often.

> If we can figure out the migration name - which clearly we can - we should do so, instead of adding more burden on the user. This is already not the best experience for the end user.

I'm confused with 'we' vs 'end user' here. Are you saying that we'd rather burden the developer to pass in some version of `__file__` than make it harder for the end user to figure out what command needs to be run?

Reply:
> Are you saying that we'd rather burden the developer to pass in some version of __file__ than make it harder for the end user to figure out what command needs to be run?

Yes, that's what I'm saying.

Reply:
I still think passing `__file__` is better because it makes the call simpler, but my feelings are not strong enough to block the PR on account of it. I agree with Jenny that the message should display the migration name. Fine to follow your preference on how to do that.